torch/utils/data/dataset.pyi.in

```diff
@@ -0,0 +1,78 @@
+# This base template ("dataset.pyi.in") is generated from mypy stubgen with minimal editing for code injection
```

Comment:
Why don't we just provide interface stub for the functional API of `IterableDataset`? The rest of object seems not useful because PyCharm should be able to find the methods for these objects.

Reply:
Off the top of my head, I believe `mypy` uses the `.pyi` file when present and it seems to throw many errors if the interfaces for other class/objects of `dataset.py` are not present in `dataset.pyi`. I can update the comment to reflect that.


https://mypy.readthedocs.io/en/stable/stubs.html

Relevant excerpt:
""
If a directory contains both a .py and a .pyi file for the same module, the .pyi file takes precedence. This way you can easily add annotations for a module even if you don’t want to modify the source code. This can be useful, for example, if you use 3rd party open source libraries in your program (and there are no stubs in typeshed yet).
"""

Reply:
I see. This is annoying, then leave it as is.
If we want to make `IterDataPipe` inheriting `IterableDataset` rather than aliasing in the future, then we can spin off `IterDataPipe` to a new file and make a separate `pyi`.